utils/parametric/spec/trace.py

```diff
@@ -154,107 +154,37 @@ def decode_v06_stats(data: bytes) -> V06StatsPayload:
     )
 
 
-def _span_similarity(s1: Span, s2: Span) -> int:
-    """Return a similarity rating for the two given spans."""
-    score = 0
-
-    for key in set(s1.keys() & s2.keys()):
-        if s1[key] == s2[key]:
-            score += 1
-
-    s1_meta = s1.get("meta", {})
-    s2_meta = s2.get("meta", {})
-    for key in set(s1_meta.keys()) & set(s2_meta.keys()):
-        if s1_meta[key] == s2_meta[key]:
-            score += 1
-
-    s1_metrics = s1.get("metrics", {})
-    s2_metrics = s2.get("metrics", {})
-    for key in set(s1_metrics.keys()) & set(s2_metrics.keys()):
-        if s1_metrics[key] == s2_metrics[key]:
-            score += 1
-    return score
-
-
-def find_similar_trace_by_root(traces: List[Trace], span: Span) -> Trace:
-    """Return the trace from `traces` with root span most similar to `span`."""
-    assert len(traces) > 0
-
-    max_similarity = -math.inf
-    max_score_trace = traces[0]
-    for trace in traces:
-        root = root_span(trace)
-        similarity = _span_similarity(root, span)
-        if similarity > max_similarity:
-            max_score_trace = trace
-            max_similarity = similarity
-    return max_score_trace
-
-
-def find_trace_by_root(traces: List[Trace], span: Span) -> Trace:
+def find_trace(traces: List[Trace], trace_id: int) -> Optional[Trace]:
     """Return the trace from `traces` with root span matching all fields of `span`."""
```

Comment:
I think this comment doesn't match the function anymore.
```suggestion
    """Return the trace from `traces` with trace id matching `trace_id`."""
```